internal/services/compute/encryption_settings.go

```diff
@@ -13,15 +13,15 @@ func encryptionSettingsSchema() *pluginsdk.Schema {
 		MaxItems: 1,
 		Elem: &pluginsdk.Resource{
 			Schema: map[string]*pluginsdk.Schema{
+				// TODO: remove in 3.0
 				"enabled": {
-					Type:     pluginsdk.TypeBool,
-					Required: true,
-
-					// Azure can change enabled from false to true, but not the other way around, so
-					//   to keep idempotency, we'll conservatively set this to ForceNew=true
-					ForceNew: true,
+					Type:       pluginsdk.TypeBool,
+					Optional:   true,
+					Default:    true,
```

Comment:
Set the default value to `true`, so that it fits the logic when only `disk_encryption_key` is specified and `enabled` is not specified